articles/quickstart/spa/auth0-react/01.md

```diff
@@ -0,0 +1,297 @@
+
+## Prerequisites
+
+- If you don't have an Auth0 account already, <a href="https://auth0.com/signup" data-amp-replace="CLIENT_ID" data-amp-addparams="anonId=CLIENT_ID(cid-scope-cookie-fallback-name)">sign up for a free Auth0 account</a>, and [create a tenant](https://auth0.com/docs/getting-started/create-tenant).
+
+- [Register your React application](https://auth0.com/docs/dashboard/guides/applications/register-app-spa) as a single-page application with Auth0.
+
+## Install the SDK
+
+Run the following command within your project directory to install the Auth0 React SDK:
+
+```bash
+npm install @auth0/auth0-react
+```
+
+> You may use `yarn` instead.
+
+You can use the Auth0 React SDK to add authentication to your React app quickly, gain access to user profile information, and securely call APIs.
+
+## Configure the SDK
+
+Under the hood, the Auth0 React SDK uses [React Context](https://reactjs.org/docs/context.html) to manage the authentication state of your users and to expose helpful utility methods. One way to integrate Auth0 with your React app is to wrap your root component with an `Auth0Provider` that you can import from the SDK.
+
+```javascript
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+
+import {Auth0Provider} from "@auth0/auth0-react";
+
+ReactDOM.render(
+  <Auth0Provider 
+    domain="${account.namespace}"
+    client_id="${account.clientId}"
+    redirect_uri={window.location.origin}
+  >
+    <App />
+  </Auth0Provider>,
+  document.getElementById("root")
+);
+```
+
+The `Auth0Provider` component takes three props:
+ 
+- `domain` and `client_id`: The values of these properties correspond to the "Domain" and "Client ID" values present under the "Settings" of the single-page application that you registered with Auth0.
+
+- `redirect_uri`: The URL to where you'd like to redirect your users after they authenticate with Auth0. 
+
+Once configured, `Auth0Provider` will store the authentication state of your users, the state of the Auth0 Client (is Auth0 ready to use?), and provide you with helper methods to log in and log out your users. You can access the content of the `Auth0Provider` with `useAuth0()`, which is a custom [React Hook](https://reactjs.org/docs/hooks-reference.html) exposed by the SDK.
+
+## Add Login and Logout Buttons to Your App
+
+The Auth0 React SDK makes it easy to create functionality to log in or log out your users from your React application, such as a button or link. After you have configured the `Auth0Provider`, you can use the `useAuth0()` custom React Hook within any component where you need this functionality by using the following boolean flags and methods it exposes:
+
+- `isAuthenticated`: Returns `true` if the Auth0 authentication server has authenticated the user.
+
+- `isLoading`: Returns `true` if the SDK has created an Auth0 Client and has determined the user's authentication status.
```

Comment:
If the user's authentication status **has been determined**, should the value not be false? Is "isLoading" not set to true when the user is still signing in?

Reply:
True! this is a typo.

Reply:
Yeah, this is a typo. This is why I think it's more helpful for this to be called something like `isReady` instead. It's better from a cognitive and development perspective to think in "positive", `true` terms. I feel it's better to check is Auth0 `isReady` rather than if it `isLoading`.